### PR TITLE
docs: remove auto_bootstrap option from the documentation

### PR DESCRIPTION
**Pull request by annastuchlik**
Fixes https://github.com/scylladb/scylladb/issues/12318

This PR removes all occurrences of the `auto_bootstrap` option in the docs. 
In most cases, I've simply removed the option name and its definition, but sometimes additional changes were necessary:
- In node-joined-without-any-data.rst, I removed the `auto_bootstrap `option as one of the causes of the problem.
- In rebuild-node.rst, I removed the first step in the procedure (enabling the `auto_bootstrap `option).
- In admin. rst, I removed the section about manual bootstrapping - it's based on setting `auto_bootstrap` to false, which is not possible now.

### CODE REVIEW

File: docs/operating-scylla/procedures/cluster-management/rebuild-node.rst [hunk 1]
```diff
@@ -10,14 +10,9 @@ This is also why it's highly recommended to use a replication factor of at least
...
 
-#. If ``auto_bootstrap`` disabled, enable it by setting the value to ``true`` in the scylla.yaml.
-   Auto-bootstrap is enabled by default. To disable it, a setting is normally added to the scylla.yaml.
-   The Scylla configuration file is located in ``/etc/scylla/scylla.yaml``.
-   Look for ``auto_bootstrap: false``. If this setting is in the file, change it to true.
+#. Open the ``/etc/scylla/scylla.yaml`` file.
 
-   .. note:: If the ``auto_bootstrap`` parameter is missing from ``/etc/scylla/scylla.yaml``, default setting ``true`` is applied.
-
-#. With the yaml file still open, add, if not present,else edit, the ``replace_address_first_boot`` parameter and change it to the
```
Comment (kbr-scylla) [inline docs/operating-scylla/procedures/cluster-management/rebuild-node.rst:20]: There's a second mention of `auto_bootstrap` below.

File: docs/troubleshooting/node-joined-without-any-data.rst [hunk 2]
```diff
@@ -11,12 +11,10 @@ Use the procedure that follows to correct situations where a node joins a cluste
 The Problem
 ^^^^^^^^^^^
 
-There are two possible causes for a new node not acquiring data from its peers:
+A new node may not acquire data from its peers if he ``seeds`` parameter in the new node's ``scylla.yaml`` file
```
Comment (kbr-scylla) [inline docs/troubleshooting/node-joined-without-any-data.rst:14]: typo: "he" (should be "the")

**Comment by kbr-scylla** [comment]
BTW. removing "auto_bootstrap" from docs is by itself a sufficiently "atomic" change, no need to split it to separate commits. For example it wouldn't make much sense to backport some of these commits and not others.
(Although it doesn't hurt, if you prefer the separate commits then it's fine)

**Comment by annastuchlik** [comment]
Thanks! I've made the changes you pointed out. Can you re-review?

I decided to split the PR into separate commits because sometimes the change was about rewriting/removing the paragraph, and I wasn't sure it was the right thing to do.

**Comment by kbr-scylla** [comment]
The final state looks good.

However, please clean up the git log - instead of putting the changes from review into a new commit, modify the earlier commits, including changes from that last commit where necessary: the changes to rebuild-node.rst should be included in the [docs: remove the auto_bootstrap option from the procedure to rebuild...](https://github.com/scylladb/scylladb/pull/12419/commits/5ca71ca343665297c788eda1e7cc5475fdd79c26) commit and so on.

If you haven't done this before, I recommend using `git rebase -i` to edit the earlier commits and `git checkout -p` to include the appropriate changes from the last commit:
1. `git checkout` the branch
2. Make a backup branch just in case (`git branch <backup-branch-name>`)
3. `git rebase -i <base-commit>` where `<base-commit>` is the commit on which you based this branch (currently 8635d24424dab59e4054209ec3f2616a246a9282)
4. Your editor will open with the list of commits on your branch and operations that you want to perform on each commit; change `pick` to `edit` for the commits that you want to edit (e.g. the rebuild commit)
5. Save and close the file, interactive rebase will start, git will now checkout the first commit that you selected for editing
6. `git checkout -p <last-commit>`, where `<last-commit>` is the last commit in the branch (currently 3200ff7635b185fb6ab1c00a0d33b5ad5c41926f)
7. git will show you hunks and ask which ones you want to include. Include the ones that should be part of the commit that rebase is currently editing (`y`) and drop the others (`n`). If a hunk is too large you can split it using `s`.
8. After `git checkout -p` finishes, confirm that you included all the desired changes with `git status` and `git diff --staged`.
9. `git rebase --continue` to go to the next commit that you selected for editing. When you do `git rebase --continue` on the last selected commit, the rebase will finish.

During rebase you can always `git rebase --abort` if you make a mistake.

**Comment by annastuchlik** [comment]
@kbr-scylla Thanks for the instructions! I think it worked 😌

**Review by kbr-scylla — APPROVED** [review]
Nice.

**Pull request merged**